rtgui/edit.cc

```diff
@@ -747,7 +747,294 @@ void Rectangle::drawToMOChannel(Cairo::RefPtr<Cairo::Context> &cr, unsigned shor
+{
+    // test of F_HOVERABLE has already been done
+
+    int imgW = img->get_width();
+    int imgH = img->get_height();
+
+    rtengine::Coord pos;
+
+    if (datum == IMAGE)
+        coordSystem.imageCoordToCropCanvas (position.x, position.y, pos.x, pos.y);
+    else if (datum == CLICKED_POINT) {
+        pos = position + objectBuffer->getDataProvider()->posScreen;
+    } else if (datum == CURSOR)
+        pos = position + objectBuffer->getDataProvider()->posScreen
+              + objectBuffer->getDataProvider()->deltaScreen;
+
+    rtengine::Coord tl, br; // Coordinate of the rectangle in the CropBuffer coordinate system
+    drivenPointToRectangle(pos, tl, br, imgW, imgH);
+
+    // drawing the lower byte's value
+    if (objectBuffer->getObjectMode() == OM_255) {
+        cr->set_source_rgba (0., 0., 0., ((id + 1) & 0xFF) / 255.);
+    } else {
+        cr->set_source_rgba (0., 0., 0., (id + 1) / 65535.);
+    }
+    cr->set_line_width(0.);
+    cr->rectangle(tl.x, tl.y, imgW, imgH);
+    cr->fill();
+}
+
+void OPIcon::drawOuterGeometry(Cairo::RefPtr<Cairo::Context> &cr,
+                               ObjectMOBuffer *objectBuffer, EditCoordSystem &coordSystem) {}
+
+void OPIcon::drawInnerGeometry(Cairo::RefPtr<Cairo::Context> &cr,
+                               ObjectMOBuffer *objectBuffer, EditCoordSystem &coordSystem)
+{
+    if (flags & F_VISIBLE) {
+        // Here we will handle fallback solutions
+        State tmpState = state;
+
+        if (tmpState == INSENSITIVE) {
+            if (!insensitiveImg) {
+                tmpState = NORMAL;
+            } else {
+                OPIcon::drawImage(insensitiveImg, cr, objectBuffer, coordSystem);
+                return;
+            }
+        }
+
+        if (tmpState == DRAGGED) {
+            if (!draggedImg) {
+                tmpState = ACTIVE;
+            } else {
+                OPIcon::drawImage(draggedImg, cr, objectBuffer, coordSystem);
+                return;
+            }
+        }
+
+        if (tmpState == ACTIVE) {
+            if (!activeImg) {
+                tmpState = PRELIGHT;
+            } else {
+                OPIcon::drawImage(activeImg, cr, objectBuffer, coordSystem);
+                return;
+            }
+        }
+
+        if (tmpState == PRELIGHT) {
+            if (!prelightImg) {
+                tmpState = NORMAL;
+            } else {
+                OPIcon::drawImage(prelightImg, cr, objectBuffer, coordSystem);
+                return;
+            }
+        }
+
+        if (tmpState == NORMAL && normalImg) {
+            OPIcon::drawImage(normalImg, cr, objectBuffer, coordSystem);
+        }
+    }
+}
+
+void OPIcon::drawToMOChannel(Cairo::RefPtr<Cairo::Context> &cr, unsigned short id,
+                             ObjectMOBuffer *objectBuffer, EditCoordSystem &coordSystem)
+{
+    if (flags & F_HOVERABLE) {
+        // Here we will handle fallback solutions
+        State tmpState = state;
+
+        if (tmpState == INSENSITIVE) {
```

Comment:
Are these not exclusive? I.e. shouldn't this be `if-else-if-else` or even `switch-case`?